### PR TITLE
Ensure that @QuarkusTest plays nicely with @testfactory

### LINKED ISSUE #8004: Regression: @Inject mock service in QuarkusTest is null during JUnit @TestFactory execution in Quarkus 1.3.0.Final
**Description**
When a JUnit test includes an `@Inject` mock service bean initialisation works properly for normal `@BeforeEach` and `@Test` annotated methods, but tests executed with `@TestFactory` have the mock service uninitialised.

In the attached sample project, look at ExampleResourceTest.`tests` for an example of the problem, these tests work properly in Quarkus 1.2.1.Final.

**Expected behavior**
The `@Inject` mock bean is initialised correctly during `@TestFactory` execution.

**Actual behavior**
The `@Inject` mock bean fails to be initialised.

`
[ERROR] tests[1]  Time elapsed: 0.154 s  <<< ERROR!
kotlin.UninitializedPropertyAccessException: lateinit property mockExternalService has not been initialized
        at org.acme.ExampleResourceTest.getMockExternalService(ExampleResourceTest.kt:13)
`

**To Reproduce**
1. In the attached sample project, run ExampleResourceTest.`tests`
2. kotlin.UninitializedPropertyAccessException: lateinit property mockExternalService has not been initialized

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
`Darwin xxxx 18.7.0 Darwin Kernel Version 18.7.0: Thu Jan 23 06:52:12 PST 2020; root:xnu-4903.278.25~1/RELEASE_X86_64 x86_64`
 - Output of `java -version`: 
`
$ java -version
java version "1.8.0_172"
Java(TM) SE Runtime Environment (build 1.8.0_172-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.172-b11, mixed mode)
`
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 
`1.3.0.Final`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
`
$ ./mvnw --version                                                                                                                                                                                 11:19:03
Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-05T06:00:29+11:00)
Maven home: /Users/xxxx/.m2/wrapper/dists/apache-maven-3.6.1-bin/38pn40mp89t5c94bjdbeod370m/apache-maven-3.6.1
Java version: 1.8.0_172, vendor: Oracle Corporation, runtime: /Library/Java/JavaVirtualMachines/jdk1.8.0_172.jdk/Contents/Home/jre
Default locale: en_AU, platform encoding: UTF-8
OS name: "mac os x", version: "10.14.6", arch: "x86_64", family: "mac"
`
**Additional context**
[Sample project](https://github.com/quarkusio/quarkus/files/4357414/code-with-quarkus.zip)

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: core/runtime/pom.xml
Comment (gastaldi) [inline core/runtime/pom.xml]: This is just a pom.xml depending on `api`, `params` and `engine` to save users from adding them explicitly in their poms. Is it necessary here?

Comment (stuartwdouglas) [reply core/runtime/pom.xml]: In that case no.

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
@stuartwdouglas @gastaldi I'll let you decide if you want to tweak the pom change or not :)

**Review by gastaldi — APPROVED** [review]
I removed the extra tag. Ready to merge when CI finishes

**Pull request merged**
